controller/resource.go

```diff
@@ -259,14 +243,15 @@ func (s *ServiceInfo) Upsert() error {
 	}
 
 	// Otherwise, create one.
-	err = s.updateServiceToken(secretName)
+	err = s.createServiceToken(currSecret)
 	if err != nil {
 		return fmt.Errorf("updating service token: %w", err)
 	}
 
 	return nil
 }
 
+// Delete removes the token for the given ServiceInfo.
 func (s *ServiceInfo) Delete() error {
```

Comment:
I've considered making this change, but am not 100% sure it is an overall positive. I'm imagining a scenario where Consul is in a bad state and fails to process a ACL token deletion. Does it make sense to continue deleting tokens when we know that it [will retry in 10 seconds](https://github.com/hashicorp/consul-ecs/blob/main/controller/controller.go#L11) anyway? Also, the loop here is just for the case where two ACL tokens have been generated by the ACL controller for the same service which I think will only happen if two ACL controllers are running and making tokens for the same service at the same time.

I'm not sure if others agree, but my overall philosophy for the ACL controller is to give up if anything goes wrong because insuring the stability of Consul is more important than making tokens faster.

What do you think?

Reply:
At the very least, it feels like we should backlog adding a README for the ACL controller that describes how it works and it's overall philosophy.